File: 2025_07/leetcode2025-07-28.cpp
#include <iostream>
#include <vector>
#include <algorithm>
#include <cmath>
using namespace std;

class Solution {
public:
    int countMaxOrSubsets(vector<int>& nums) {
        int n = nums.size();
        int mx = 0, cnt = 0;
        for (int i=0; i<pow(2, n); i++){
            int cur = 0;
            for (int j=0; j<n; j++){
                if ((i>>j)&1){
                    cur |= nums[j];
                }
            }
            if (mx < cur){
                mx = cur;
                cnt = 1;
            }
            else if (mx == cur){
                cnt++;
            }
        }
        return cnt;
    }
};

int main(){
    Solution s;
    vector<int> nums = {3,1};
    cout << s.countMaxOrSubsets(nums) << endl;
}

File: leetcode2025/2025_07/leetcode2025-07-28.py
from typing import List

class Solution:
    def countMaxOrSubsets(self, nums: List[int]) -> int:
        n = len(nums)
        cnt = 0
        mx = 0
        def dfs(indx, val):
            if indx == n:
                nonlocal mx,cnt
                if mx < val:
                    mx = val
                    cnt = 1
                elif mx == val:
                    cnt += 1
                return
            dfs(indx+1, val|nums[indx])
            dfs(indx+1, val)
        dfs(0, 0)
        return cnt

if __name__ == "__main__":
    s = Solution()
    nums = [3,1]
    print(s.countMaxOrSubsets(nums))
